website/profile/views.py

```diff
@@ -521,12 +529,40 @@ def update_mailchimp_subscription(user, list_name, subscription, send_goodbye=Tr
             pass
     else:
         try:
-            mailchimp_utils.unsubscribe_mailchimp_async(list_name, user._id, username=user.username, send_goodbye=send_goodbye)
+            mailchimp_utils.unsubscribe_mailchimp_async(list_name, user._id, username=user.username)
         except (MailChimpError, OSFError):
             # User has already unsubscribed, so nothing to do
             pass
 
 
+@user_account_merged.connect
+def send_account_merged_message(user):
+    """ Sends a message using Celery messaging to alert other services that an osf.io user has been merged."""
+    publish_merged_user(user)
+
+
+@user_account_merged.connect
+def unsubscribe_old_merged_account_from_mailchimp(user):
+    """ This is a merged account (an old account that was merged into an active one) so it needs to be unsubscribed
+    from mailchimp."""
+    for key, value in user.mailchimp_mailing_lists.items():
+        subscription = value or user.merged_by.mailchimp_mailing_lists.get(key)
+        update_mailchimp_subscription(user.merged_by, list_name=key, subscription=subscription)
+        update_mailchimp_subscription(user, list_name=key, subscription=False)
```

Comment:
I'm not sure this needs to exist, for two reasons -- 
1) The `unsubscribe_mailchimp_async` is still being called above, and
2) `blinker.signal`s aren't async by default (requires [async receivers](https://blinker.readthedocs.io/en/stable/#async-receivers) ), which may lead to this either causing longer request cycles (blocking on a MailChimp call) or ignorable errors (e.g. "User already unsubscribed") causing these transactions to fail.

Reply:
I'm a little confused about these comments because:
1. unsubscribing was happening in the second signal here: https://github.com/CenterForOpenScience/osf.io/pull/10565/files#diff-7bc4865544e245703bae45395b7fba97ab991bedc070c25697305c6d7ce5c9aaL756 so I'm not sure where you mean.
2. These are were initially were already blinker signals for `user_merged` so it's deliberately going from signal to signal.

Reply:
Ah, you're right, I misread and got confused about the call chain.

However, do we want the `merged_by` user to inherit the subscription status of the mergee? That seems like a change in behavior